torch/lib/c10d/ProcessGroupMPI.cpp

```diff
@@ -137,6 +137,81 @@ const std::exception& ProcessGroupMPI::WorkMPI::exception() const {
   }
 }
 
+ProcessGroupMPI::AsyncWork::AsyncWork(
+    at::Tensor tensor,
+    MPI_Request request,
+    int* srcRank)
+    : tensor_(std::move(tensor)),
+      request_(request),
+      srcRank_(srcRank) {
+  memset(&status_, 0, sizeof(status_));
+}
+
+ProcessGroupMPI::AsyncWork::~AsyncWork() {
+  if (request_ != MPI_REQUEST_NULL) {
+    throw std::runtime_error(
+        "Invalid call to AsyncWork::~AsyncWork before work has completed");
```

Comment:
Better not to put the destructor name in the user error message, let's rephrase it

Reply:
Done